### ISSUE TITLE
Course smoothing.

### ISSUE THREAD

**Issue by WhiskeyBravo813** [Courseplay/courseplay#4172]
I've noticed that on some fields, particularly fields with jagged edges, CP AI tries so hard to follow the precise contours of the field edge that they constantly stop and go.  This results in a very jerky movement.  I've noticed it mostly on headland tracks but if I try to run a spiral, headland-only route, those irregular lines transfer from track to track.

Would it be possible for CP to smooth out those tracks?  Perhaps averaging the course over 5 or 10 waypoints.

**Comment by Tensuko** [comment]
Screenshot ?

**Comment by pvaiko** [comment]
Is this happening with 287 or earlier too?

**Comment by mdbguy** [comment]
As a workaround for this (at least since about version 238) I have been making a few changes:
cp.lua (~ line 82)
	local minDistanceBetweenPoints = 5

center.lua (~ line 69)
	courseGenerator.waypointDistance = 15

coursegenerator.lua (~ line 5)
	courseGenerator.waypointDistance = 15

**Comment by pvaiko** [comment]
@WhiskeyBravo813 @mdbguy what map/field is this? Do you have a screenshot of that course? I really have a hard time believing the "constantly stop and go" and that workaround really shouldn't be necessary, there must be something else going on here.

**Comment by mdbguy** [comment]
map example is GWExtreme.  Field edges are very messy so understand that ideally that would be fixed first.
Basic scenario in my case - using multiple large tools (3x 16m) i get jumpy driving behaviour described above using default minDistanceBetweenPoints = 0.5
![fsScreen_3](https://user-images.githubusercontent.com/32879507/63947295-0794de00-ca4d-11e9-8ed3-9f4bbc92e629.png)
Can see the very close waypoints and resulting speed of tractor.

using minDistanceBetweenPoints = 5
![fsScreen_4](https://user-images.githubusercontent.com/32879507/63947568-8722ad00-ca4d-11e9-8300-f8033b1ac6ca.png)

**Comment by pvaiko** [comment]
Still did not get an answer: is this happening with 287 or earlier too?

**Comment by mdbguy** [comment]
Earlier versions too.   ("As a workaround for this (at least since about version 238) ")

**Comment by pvaiko** [comment]
Even with that messy field edge there really shouldn't be any "stop and go" (unless maybe with versions 288-293). Do you happen to have a video of that, with debug channels 12 and 1?f

**Comment by WhiskeyBravo813** [comment]
I don't have a way to capture video at the moment.  Also, to clarify, it
wasn't so much stop and go as just very jerky, constantly slowing every
time it needed to turn.  It seemed like every time the CP AI detects a
course change, there's a reduction in speed, as if the turn speed was
called.

That said, I just ran a fertilizer course with the latest CP release
(.00294) and didn't notice the issue.  (Also, the 3D course rendering
doesn't kill my framerates any more either, so that's good.)

If I see the issue again on another field/map, I'll see what I can do to
capture a few seconds of video and upload it.

WB813
